test/cluster/dynamic_config_test.go

```diff
@@ -607,6 +607,119 @@ var _ = Describe(
 				)
 			},
 		)
+
+		Context(
+			"When changing fields those need recluster", func() {
+				BeforeEach(
+					func() {
+						// Create a 4 node cluster
+						aeroCluster := createNonSCDummyAerospikeCluster(
+							clusterNamespacedName, 4,
+						)
+						aeroCluster.Spec.Storage.Volumes = append(
+							aeroCluster.Spec.Storage.Volumes,
+							asdbv1.VolumeSpec{
+								Name: "ns1",
+								Source: asdbv1.VolumeSource{
+									PersistentVolume: &asdbv1.PersistentVolumeSpec{
+										Size:         resource.MustParse("1Gi"),
+										StorageClass: storageClass,
+										VolumeMode:   v1.PersistentVolumeBlock,
+									},
+								},
+								Aerospike: &asdbv1.AerospikeServerVolumeAttachment{
+									Path: "/test/dev/xvdf1",
+								},
+							},
+						)
+						aeroCluster.Spec.EnableDynamicConfigUpdate = ptr.To(true)
+						aeroCluster.Spec.Image = "aerospike.jfrog.io/docker/aerospike/aerospike-server-enterprise-rc:7.2.0.0-rc2"
+						aeroCluster.Spec.PodSpec.ImagePullSecrets = []v1.LocalObjectReference{
+							{
+								Name: "regcred",
+							},
+						}
+						aeroCluster.Spec.RackConfig.Racks = append(aeroCluster.Spec.RackConfig.Racks,
+							asdbv1.Rack{
+								ID: 1,
+							},
+							asdbv1.Rack{
+								ID: 2,
+							})
+						aeroCluster.Spec.RackConfig.Namespaces = []string{
+							"test",
+							"test1",
+						}
+						nsList := aeroCluster.Spec.AerospikeConfig.Value["namespaces"].([]interface{})
+						nsList = append(nsList, getSCNamespaceConfig("test1", "/test/dev/xvdf1"))
+						aeroCluster.Spec.AerospikeConfig.Value["namespaces"] = nsList
+						err := deployCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+					},
+				)
+
+				AfterEach(
+					func() {
+						aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						_ = deleteCluster(k8sClient, ctx, aeroCluster)
+					},
+				)
+
+				It(
+					"Should update active-rack dynamically", func() {
+
+						By("Modify dynamic config by adding fields")
+
+						aeroCluster, err := getCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						podPIDMap, err := getPodIDs(ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						nsList := aeroCluster.Spec.AerospikeConfig.Value["namespaces"].([]interface{})
+						nsList[0].(map[string]interface{})["active-rack"] = 1
+						nsList[1].(map[string]interface{})["active-rack"] = 2
+
+						err = updateCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Fetch and verify dynamic configs")
+
+						pod := aeroCluster.Status.Pods["dynamic-config-test-1-0"]
+
+						info, err := requestInfoFromNode(logger, k8sClient, ctx, clusterNamespacedName, "namespace/test", &pod)
```

Comment:
I think, this kind of getInfo and parsing the info would be available in management lib. Just crosscheck

Reply:
Decided to enhance the management lib in the future and use that.